### PR TITLE
Add kms signer for `staking-update.js` and add initial metadata script

### PR DESCRIPTION
**Pull request by shark0der**
## Context

Staking update script should be run as swap controller so we need to use aws kms signer.
Staking pools metadata was migrated to staking products so we needed a script for the tx data.

## Changes proposed in this pull request

--

## Test plan

Please describe the tests cases that you ran to verify your changes. Add further instructions on 
how to run them if needed (i.e. migration / deployment scripts, env vars, etc).

## Checklist

- [ ] Rebased the base branch
- [ ] Attached corresponding Github issue
- [ ] Prefixed the name with the type of change (i.e. feat, chore, test)
- [ ] Performed a self-review of my own code
- [ ] Followed the style guidelines of this project
- [ ] Made corresponding changes to the documentation
- [ ] Didn't generate new warnings
- [ ] Didn't generate failures on existing tests
- [ ] Added tests that prove my fix is effective or that my feature works

## Review

When reviewing a PR, please indicate intention in comments using the following emojis:
* :cake: = Nice to have but not essential.
* :bulb: = Suggestion or a comment based on personal opinion
* :hammer: = I believe this should be changed.
* :thinking: = I don’t understand something, do you mind giving me more context?
* :rocket: = Feedback

### CODE REVIEW

File: scripts/staking-set-initial-pools-metadata.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+async function main() {
+  const factory = await ethers.getContractAt(StakingPoolFactory, addresses.StakingPoolFactory);
+  const stakingPoolCount = (await factory.stakingPoolCount()).toNumber();
+  const stakingPoolIds = new Array(stakingPoolCount).fill('').map((_, i) => i + 1);
+
+  const ipfsHashes = [];
+
+  for (const poolId of stakingPoolIds) {
+    const cover = await ethers.getContractAt('Cover', addresses.Cover);
```
Comment (rackstar) [inline scripts/staking-set-initial-pools-metadata.js:12]: nit: its better its this is outside the loop

File: scripts/staking-set-initial-pools-metadata.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  const encodedData = ethers.utils.defaultAbiCoder.encode(['string[]'], [ipfsHashes]);
+  const functionSignature = ethers.utils.id('setInitialMetadata(string[])').slice(0, 10);
+  const data = functionSignature + encodedData.slice(2);
+
+  console.log('Tx details:');
+  console.log('from: [any ab member]');
+  console.log('to:', addresses.StakingProducts);
+  console.log('msg.data', data);
```
Comment (rackstar) [inline scripts/staking-set-initial-pools-metadata.js:32]: We should simulate this to tenderly then we'll send the successful simulated tx to the the AB member so they can copy the tx.data to execute themselves.

This helpers file has `simulateTransaction` function that we can use:
https://github.com/NexusMutual/smart-contracts/blob/release-candidate/scripts/products/helpers.js

**Comment by rackstar** [comment]
@shark0der we have a fork test for staking-update to execute on release. is it worth have one too for metadata?

File: scripts/staking-set-initial-pools-metadata.js (hunk 2 above)
Comment (shark0der) [reply scripts/staking-set-initial-pools-metadata.js:32]: manual simulation should do it. the ab will copy the tx data manually anyway

Comment (rackstar) [reply scripts/staking-set-initial-pools-metadata.js:32]: ok, i'll add steps in the release docs for the migration plan - including this manual tenderly simulation

**Comment by rackstar** [comment]
> @shark0der we have a fork test for staking-update to execute on release. is it worth have one too for metadata?

for visibility:

fork test will be skipped in favour of tenderly simulation on the output of `staking-set-initial-pools-metadata.js`

**Review by rackstar — APPROVED** [review]

**Pull request merged**